### PR TITLE
ibm-sw-tpm2: add Darwin support

### PR DESCRIPTION
**Pull request by stv0g**
## Description of changes

The upstream sources ship with a dedicated Makefile for macOS. This commit conditionally uses the macOS makefile if build on Darwin.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by delroth** [comment]
@ofborg build ibm-sw-tpm2

**Review by delroth — APPROVED** [review]
Seems to build fine. Waiting for eval to complete because I can't remember if this needs to go to staging or not (I think it might be a checkInputs of tpm2-tss which is a dep of systemd...)

File: pkgs/tools/security/ibm-sw-tpm2/default.nix [hunk 1]
```diff
@@ -30,12 +35,14 @@ stdenv.mkDerivation rec {
 
   sourceRoot = "src";
 
+  inherit makefile;
+
```
Comment (wegank) [inline pkgs/tools/security/ibm-sw-tpm2/default.nix:39]:
```suggestion
```

Comment (stv0g) [reply pkgs/tools/security/ibm-sw-tpm2/default.nix:39]: Are you suggesting to remove the inherit statement? I am still a newcomer to Nix, but I have a hard time understanding how this should work...

Comment (wegank) [reply pkgs/tools/security/ibm-sw-tpm2/default.nix:39]: Yes.

Comment (delroth) [reply pkgs/tools/security/ibm-sw-tpm2/default.nix:39]: I don't get it either @wegank. How would that even work? Can you provide more feedback than just one word when reviewing code and making such suggestions?

https://nixos.org/manual/nixpkgs/stable/#var-stdenv-makefile

**Review by delroth — CHANGES_REQUESTED** [review]
This change causes a large amount of rebuilds (I was suspecting this would be the case, see above), thus it needs to be rebased to staging: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

File: pkgs/tools/security/ibm-sw-tpm2/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/tools/security/ibm-sw-tpm2/default.nix:39]: Oh, I'm completely wrong, sorry.

**Comment by wegank** [comment]
> This change causes a large amount of rebuilds (I was suspecting this would be the case, see above), thus it needs to be rebased to staging: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

This won't be necessary if `inherit makefile;` is made conditional on darwin, I think? Something like adding `// lib.optionalAttrs stdenv.isDarwin { inherit makefile; }` to the end of the file.

**Comment by delroth** [comment]
I think I'd rather have this go to staging and have a clean / more maintainable derivation for the future, rather than accumulating hacks like this to try and reduce rebuilds. I'm not 100% against your suggestion, but I don't think that in the case of this PR specifically anything warrants *not* going through staging.

**Review by delroth — APPROVED** [review]

**Pull request merged**